conda_build/pypi.py

```diff
@@ -287,15 +287,39 @@ def main(args, parser):
         if is_url:
             d['version'] = 'UNKNOWN'
         else:
+            versions = client.package_releases(package, True)
+            if args.version_compare:
+                from os.path import abspath, isdir
+                from conda_build.metadata import MetaData
+                recipe_dir = abspath(package)
+                if not isdir(recipe_dir):
+                    sys.exit("Error: no such directory: %s" % recipe_dir)
+                m = MetaData(recipe_dir)
+                local_version = m.version()
+                print("Local recipe for %s has version %s" % (package,
+                      local_version))
+                if local_version not in versions:
+                    sys.exit("Error: %s %s is not available on PyPI."
+                             % (package, local_version))
+                else:
+                    new_versions = [ver for ver in versions
+                                    if ver > local_version]
```

Comment:
You should use `conda.resolve.normalized_version`, which will return a version object that you can compare against. Alternately, the versions on PyPI are already in order, so just check where the current version is in the list and return the ones before it (or after it, I don't remember).